### ISSUE TITLE
Atlased textures

### ISSUE THREAD

**Issue by marrub--** [Team-Doominati/Doominati#15]

**Comment by marrub--** [comment]
to elaborate: there should be both in-memory generation and the ability to define textures as manually atlased, in case you want that

**Comment by Kyle873** [comment]
Is there some pre-defined atlas definition format we could take advantage of for manual atlasing, or would we need to develop something ourselves? I guess it could just be defined in scripts themselves or some such?

**Comment by marrub--** [comment]
Yeah, we'd just have our own definition format for pre-atlased textures, and the game would put anything that isn't pre-atlased into one of a set of atlas textures in-memory.

**Comment by Kyle873** [comment]
There's a concept from GMS that might be helpful here as well: [texture groups](https://docs.yoyogames.com/source/dadiospice/001_advanced%20use/global%20game%20settings/texture%20groups%20tab.html).

**Comment by Kyle873** [comment]
Also thinking about it, exposing natives related to allowing the user to setup the atlasing and paging themselves would be good to give the power if necessary to minimize texture swaps.

**Comment by marrub--** [comment]
I NEED THIS NOW BEFORE I FUCKING KILL VRAM AND PERFORMANCE
